internal/runtime/internal/importsource/import_git.go

```diff
@@ -194,10 +194,12 @@ func (im *ImportGit) Update(args component.Arguments) (err error) {
 	if im.repo == nil || !reflect.DeepEqual(repoOpts, im.repoOpts) {
 		r, err := vcs.NewGitRepo(context.Background(), repoPath, repoOpts)
 		if err != nil {
-			if errors.As(err, &vcs.UpdateFailedError{}) {
+			isUpdateFailedError := errors.As(err, &vcs.UpdateFailedError{})
+			if isUpdateFailedError {
 				level.Error(im.log).Log("msg", "failed to update repository", "err", err)
 				im.updateHealth(err)
-			} else {
+			}
+			if errors.As(err, &vcs.InvalidRevisionError{}) || !isUpdateFailedError {
```

Comment:
I think that with the change that was made in git.go, we can keep the original version of the code for this file because InvalidRevisionError is not an isUpdateFailedError error.

Reply:
Makes sense to me. Updated in HEAD :D